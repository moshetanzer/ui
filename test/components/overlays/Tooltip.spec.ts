import { describe, it, expect } from 'vitest'
import { UTooltip } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Tooltip', () => {
  it.each([
    ['with text', { props: { text: 'tooltip text' } }],
    ['with shortcuts', { props: { shortcuts: ['G', 'I'] } }],
    ['with default slot', { slots: { default: () => 'Default slot' } }],
    ['with text slot', { slots: { text: () => 'Text slot' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UTooltip.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UTooltip)
    expect(html).toMatchSnapshot()
  })
})
