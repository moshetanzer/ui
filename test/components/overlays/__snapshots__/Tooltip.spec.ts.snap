// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > renders with default slot correctly 1`] = `
"<div class="relative inline-flex">Default slot<div class="z-20 group max-w-xs" style="position: fixed; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom">
    <transition-stub enteractiveclass="transition ease-out duration-200" enterfromclass="opacity-0 translate-y-1" entertoclass="opacity-100 translate-y-0" leaveactiveclass="transition ease-in duration-150" leavefromclass="opacity-100 translate-y-0" leavetoclass="opacity-0 translate-y-1" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div class="[@media(pointer:coarse)]:hidden h-6 px-2 py-1 text-xs font-normal truncate relative bg-white dark:bg-gray-900 text-gray-900 dark:text-white rounded shadow ring-1 ring-gray-200 dark:ring-gray-800">
          <!--v-if-->
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Tooltip > renders with shortcuts correctly 1`] = `
"<div class="relative inline-flex"> Hover <div class="z-20 group max-w-xs" style="position: fixed; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom">
    <transition-stub enteractiveclass="transition ease-out duration-200" enterfromclass="opacity-0 translate-y-1" entertoclass="opacity-100 translate-y-0" leaveactiveclass="transition ease-in duration-150" leavefromclass="opacity-100 translate-y-0" leavetoclass="opacity-0 translate-y-1" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div class="[@media(pointer:coarse)]:hidden h-6 px-2 py-1 text-xs font-normal truncate relative bg-white dark:bg-gray-900 text-gray-900 dark:text-white rounded shadow ring-1 ring-gray-200 dark:ring-gray-800"><span class="hidden md:inline-flex flex-shrink-0 gap-0.5"><span class="mx-1 text-gray-700 dark:text-gray-200">Â·</span><kbd class="inline-flex items-center justify-center text-gray-900 dark:text-white h-4 min-w-[16px] text-[10px] px-1 rounded font-medium font-sans bg-gray-100 dark:bg-gray-800 ring-1 ring-gray-300 dark:ring-gray-700 ring-inset">G</kbd><kbd class="inline-flex items-center justify-center text-gray-900 dark:text-white h-4 min-w-[16px] text-[10px] px-1 rounded font-medium font-sans bg-gray-100 dark:bg-gray-800 ring-1 ring-gray-300 dark:ring-gray-700 ring-inset">I</kbd></span></div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Tooltip > renders with text correctly 1`] = `
"<div class="relative inline-flex"> Hover <div class="z-20 group max-w-xs" style="position: fixed; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom">
    <transition-stub enteractiveclass="transition ease-out duration-200" enterfromclass="opacity-0 translate-y-1" entertoclass="opacity-100 translate-y-0" leaveactiveclass="transition ease-in duration-150" leavefromclass="opacity-100 translate-y-0" leavetoclass="opacity-0 translate-y-1" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div class="[@media(pointer:coarse)]:hidden h-6 px-2 py-1 text-xs font-normal truncate relative bg-white dark:bg-gray-900 text-gray-900 dark:text-white rounded shadow ring-1 ring-gray-200 dark:ring-gray-800">tooltip text
          <!--v-if-->
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Tooltip > renders with text slot correctly 1`] = `
"<div class="relative inline-flex"> Hover <div class="z-20 group max-w-xs" style="position: fixed; left: 0px; top: 0px; margin: 0px; right: auto; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom">
    <transition-stub enteractiveclass="transition ease-out duration-200" enterfromclass="opacity-0 translate-y-1" entertoclass="opacity-100 translate-y-0" leaveactiveclass="transition ease-in duration-150" leavefromclass="opacity-100 translate-y-0" leavetoclass="opacity-0 translate-y-1" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div class="[@media(pointer:coarse)]:hidden h-6 px-2 py-1 text-xs font-normal truncate relative bg-white dark:bg-gray-900 text-gray-900 dark:text-white rounded shadow ring-1 ring-gray-200 dark:ring-gray-800">Text slot
          <!--v-if-->
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;
